### ISSUE TITLE
Dropdown caret is created and positioned with CSS generated content which is a WCAG failure (F87)

### ISSUE THREAD

**Issue by fstorr** [twbs/bootstrap#30727]
Dropdown elements in the [Navbar element](https://getbootstrap.com/docs/4.0/components/navbar/) use CSS generated content to create and place the caret:

![screenshot of the HTML for a dropdown link with the CSS :after pseudo element highlighted](https://user-images.githubusercontent.com/169471/80998437-d3519880-8df7-11ea-806e-310ba6da6c1d.png)

Because the caret is needed to inform users that there are dropdown items, it's non-decorative content which is covered by [WCAG's F87](https://www.w3.org/TR/WCAG20-TECHS/F87.html). ("Failure of Success Criterion 1.3.1 due to inserting non-decorative content by using :before and :after pseudo-elements and the 'content' property in CSS").

Replacing the carets with inline SVG will also close [Caret is a rectangle with Windows High Contrast Mode](https://github.com/twbs/bootstrap/issues/27650).

**Comment by ffoodd** [comment]
FWIW `aria-haspopup` and `aria-expanded` are informing users that this is a disclosure widget. Thus caret is probably decorative—however I'd like @patrickhlauke opinion too 🙂

And I think @MartijnCuppens has a draft PR about inlining SVGs, I don't know if carets are concerned yet.

**Comment by fstorr** [comment]
Hey :)

`aria-haspopup` and `aria-expanded` are there for assistive-technology users. A screen-reader user will hear the presence and state of those attributes, but others almost certainly won't know they exist.

An example of a dropdown caret being non-decorative can be seen in the Active User Interface Component Examples section of WCAG's [Non-Text Contrast Understanding document](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#active-user-interface-component-examples):

> "The down-arrow is required to understand that there is drop-down functionality …"

**Comment by patrickhlauke** [comment]
> others almost certainly won't know they exist

what kind of "others" do you foresee having a problem with CSS generated content? users who have CSS disabled?

but in short no this is not a WCAG failure.

**Comment by patrickhlauke** [comment]
and while WHCM is one scenario that deals poorly with CSS-generated content (or at least it used to), that is more a failing of WHCM's approach of haphazardly overriding/ignoring author-defined styles. not a WCAG failure per se.

lastly, citing the example for non-text contrast is orthogonal to the discussion here. not disputing that the caret is essential, but rather that using CSS-generated carets is a failure (since the concern of F87 is AT being able to still convey the information reliably, which in this case is already explicitly given programmatically with ARIA).

...and once we switch to SVG icons, the whole discussion is moot anyway
